script/tool/lib/src/create_all_packages_app_command.dart

```diff
@@ -2,49 +2,58 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'dart:io' as io;
-
 import 'package:file/file.dart';
+import 'package:meta/meta.dart';
 import 'package:path/path.dart' as p;
 import 'package:platform/platform.dart';
 import 'package:pub_semver/pub_semver.dart';
 import 'package:pubspec_parse/pubspec_parse.dart';
 
 import 'common/core.dart';
+import 'common/file_utils.dart';
 import 'common/package_command.dart';
 import 'common/process_runner.dart';
 import 'common/repository_package.dart';
 
-const String _outputDirectoryFlag = 'output-dir';
+@visibleForTesting
 
-const String _projectName = 'all_packages';
+/// The name of the build-all-packages project.
+const String allPackagesProjectName = 'all_packages';
 
-const int _exitUpdateMacosPodfileFailed = 3;
-const int _exitUpdateMacosPbxprojFailed = 4;
-const int _exitGenNativeBuildFilesFailed = 5;
+const int _exitFlutterCreateFailed = 3;
+const int _exitGenNativeBuildFilesFailed = 4;
+const int _exitMissingFile = 5;
+const int _exitMissingLegacySource = 6;
 
 /// A command to create an application that builds all in a single application.
 class CreateAllPackagesAppCommand extends PackageCommand {
   /// Creates an instance of the builder command.
   CreateAllPackagesAppCommand(
     Directory packagesDir, {
     ProcessRunner processRunner = const ProcessRunner(),
-    Directory? pluginsRoot,
     Platform platform = const LocalPlatform(),
   }) : super(packagesDir, processRunner: processRunner, platform: platform) {
-    final Directory defaultDir =
-        pluginsRoot ?? packagesDir.fileSystem.currentDirectory;
     argParser.addOption(_outputDirectoryFlag,
-        defaultsTo: defaultDir.path,
-        help:
-            'The path the directory to create the "$_projectName" project in.\n'
+        defaultsTo: packagesDir.parent.path,
+        help: 'The path the directory to create the "$allPackagesProjectName" '
+            'project in.\n'
             'Defaults to the repository root.');
+    argParser.addOption(_legacySourceFlag,
```

Comment:
No need to do this change but did you consider instead of making "legacy" a build flag taking in a root project directory that the rest of the modifications run on?

Reply:
You mean instead of combining old and new? Yes, but it means we have to maintain all the other parts that we don't care about, like the Dart code.

Reply:
Ok, I trust your judgement about the cost/value of that maintenance.